### DISCUSSION TITLE
Is it possible to pipe input?

### DISCUSSION THREAD

**Discussion by diehard2** [daniele77/cli#197] [Q&A]
I suppose this is a question/feature request. It would be super useful to be able to pipe commands or redirect to a file. Not sure how one would even go about that. Maybe overloading on ostream. Anyway, super useful library, thanks for the hard work!

**Comment by daniele77** [comment]
The library already supports pipe commands and/or redirects to a file.

For piping commands you can do, for example:

    echo "answer 42" | ./complete

Instead, for an example of taking input from a file `input.txt` and having the output on a file `output.txt` have a look at: `examples/filesession.cpp`

**Comment by diehard2** [comment]
Thanks for the reply. What I'm trying to do is

  ```
subMenu->Insert(
              "moretest", {"address:port"},
              [](std::ostream& out, const string& arg)
              {
                for(int i = 0; i < 1000; i++ ){
                 out << arg << std::endl;
                }
              },
              "Print the reverse string" );
```

if I do 

cli> moretest test | more

it would have the standard more behavior to the screen. Or 

cli> moretest test | grep foo

would return lines with "foo" in it to the screen.

**Comment by daniele77** [comment]
Well, that's a feature very hard to implement. Keep in mind that this library is not a replacement for a Linux shell, but a command prompt for your own application.

However, for the first one, you should put the pagination code inside your handler.
